### DISCUSSION TITLE
The API was successfully launched, but the API did not open any ports

### DISCUSSION THREAD

**Discussion by chettestwef** [PhlexPlexico/G5API#295] [Help]
**The configuration file has been completed, but it seems that it did not work properly as it did not open a port locally**
**The startup command I used: yarn startprod**
**Running user: root**
**Node version: 16.20.2**
![image](https://github.com/user-attachments/assets/a89b5baa-cec8-40c3-81ec-34040dcbaecd)
**The following is a screenshot of the log:**
![image](https://github.com/user-attachments/assets/159774c5-ed57-4996-ac65-16c3395844d9)

**The following is the JSON configuration. Due to some sensitive information, I have replaced some of the information with 1_2_3_4_5_6**
```
{
  "server": {
    "port": 1234,
    "hostname": "http://localhost",
    "dbKey": "d0 b1 3c 7a 1d 15 fa 0d 66 a6 80 5d b7 c7 b4 eb ",
    "steamAPIKey": "1_2_3_4_5_6",
    "sharedSecret": "1_2_3_4_5_6",
    "clientHome": "1_2_3_4_5_6",
    "useRedis": true,
    "apiURL": "http://localhost:1234/api/",
    "uploadDemos": false,
    "localLoginEnabled": true,
    "redisUrl": "redis://:1_2_3_4_5_6@localhost:6379",
    "redisTTL": 86400
  },
  "development": {
    "driver": "mysql",
    "user": "1_2_3_4_5_6",
    "password": "1_2_3_4_5_6",
    "database": "g5api",
    "multipleStatements": true,
    "flags": { "ENV": "MYSQL_FLAGS" },
    "host": "localhost",
    "port": 3306,
    "connectionLimit": 30
  },
  "admins": {
    "steam_ids": ""
  },
  "super_admins": {
    "steam_ids": "1_2_3_4_5_6,1_2_3_4_5_6"
  }
}
```
The following is a description of the problem
![image](https://github.com/user-attachments/assets/8b78a493-ee6e-45f8-bf85-828dd2117487)
![image](https://github.com/user-attachments/assets/70d8b3f6-1a04-469c-bf20-e2113b539435)
![image](https://github.com/user-attachments/assets/417f8781-6422-42ce-983c-a3fe4592e731)

**Comment by PhlexPlexico** [comment]
It's most likely failing to launch. You need to look at the logs that pm2 will provide, but I can tell from your config it's going to crash on the `dbKey` value. There cannot be any spaces, and it has to be an exact value.

It should be 
`"d0b13c7a1d15fa0d66a6805db7c7b4eb"`

This value should also be private as well, so I would recommend generating a new database key.

Thanks!
